### PR TITLE
Add slim framework support

### PR DESCRIPTION
**Pull request by Brunty**
Saw the tweet about Zend Expressive, based this off of that for support for [Slim](https://www.slimframework.com)

### CODE REVIEW

**Comment by Brunty** [comment]
Worth adding a section in the readme for a Slim application?

**Comment by ciaranmcnulty** [comment]
Good stuff! Yeah docs would be great

I'm wondering whether these should be separate extensions per framework ...

File: src/Slim/App.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
```
Comment (ciaranmcnulty) [inline src/Slim/App.php]: We haven't used strict_types elsewhere (yet)

File: tests/Slim/AppTest.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
```
Comment (ciaranmcnulty) [inline tests/Slim/AppTest.php]: We don't have this elsewhere yet

**Review by ciaranmcnulty — COMMENTED** [review]
Nothing bad to say

**Comment by ciaranmcnulty** [comment]
Two thoughts prompted by this:

 1. Oh, apps can be callable. This probably means the way that `is_callable` is checked in the CallableFactory is wrong and only working because of the order the factories are loaded into the loader!
 2. With 2 frameworks supported how will this scale? Maybe they should be different extensions (that could also do additional stuff for each framework like container injection)

File: src/Slim/App.php (hunk 1 above)
Comment (Brunty) [reply src/Slim/App.php]: Want it removed? (and from the others too?)

**Comment by Brunty** [comment]
Will add a section to the docs for Slim later.

1. Looks like it could be the case, but would mean that other Apps that are callable wouldn't work if picked up by https://github.com/ciaranmcnulty/behat-psr7extension/blob/e7448d8574d965d4c6e3811ca337e62ecf53f3ba/src/Callback/AppFactory.php#L18 as it wouldn't necessarily be providing the right arguments (unless assumed that they all take the same request/response params)
2. Extensions are probably a good idea, means that the core you've built is just the framework and extensions can have a bit more freedom to do things they need.

File: src/Slim/App.php (hunk 1 above)
Comment (ciaranmcnulty) [reply src/Slim/App.php]: Yeah just for consistency. Feel free to open a strict-all-the-things PR separately

**Comment by ciaranmcnulty** [comment]
Yeah I will add some idea of priority to the factories and make the callable one low-prority, and/or get it to use reflection and reject non-matching signatures

File: src/Slim/App.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+use Cjm\Behat\Psr7Extension\Psr7App;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Slim\App as Application;
+use Slim\Http\Response;
+
+final class App implements Psr7App
+{
+
```
Comment (ciaranmcnulty) [inline src/Slim/App.php]: [Pedantic comment about whitespace]

**Pull request merged**